core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -62,6 +69,42 @@ oppia.directive('supplementalCard', [
             $scope.helpCardHasContinueButton = false;
           };
 
+          $scope.isCurrentHintUsable = false;
+
+          $scope.allHintsExhausted = false;
+
+          $scope.promiseForHint = null;
+
+          $scope.hintButtonClicks = 0;
+
+          $scope.showHint = function() {
+            var hints = $scope.currentInteractionHints;
+            if ($scope.hintButtonClicks < hints.length) {
+              var currentHint = hints[$scope.hintButtonClicks].hintText;
+              playerTranscriptService.addNewAnswer('(Asked for a hint)');
```

Comment:
I've changed it to 'I would like a hint' for now. The variants idea seems nice, let's do that later!